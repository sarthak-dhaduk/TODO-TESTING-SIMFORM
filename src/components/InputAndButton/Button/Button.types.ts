export type ButtonProps = {
  optionHandler: () => void;
};
