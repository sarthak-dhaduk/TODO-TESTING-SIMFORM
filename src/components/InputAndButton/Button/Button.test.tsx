import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Button from "./Button";
import user from "@testing-library/user-event";

describe("Button", () => {
  //testing the ui snapshot
  test("Button renders correctly", () => {
    const renderedComponent = renderer
      .create(<Button optionHandler={() => {}} />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  //events with btn are handled
  test("Button handlers are called", async () => {
    user.setup();
    const optionHandler = jest.fn();
    render(<Button optionHandler={optionHandler} />);
    const addButton = screen.getByRole("button", {
      name: /plus/i,
    });
    await user.dblClick(addButton);
    expect(optionHandler).toHaveBeenCalledTimes(2);
  });
});
