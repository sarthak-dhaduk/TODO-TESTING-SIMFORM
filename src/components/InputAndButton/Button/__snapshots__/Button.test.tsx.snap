// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button renders correctly 1`] = `
<button
  className="absolute bottom-[calc(11vh-30px)] left-[calc(50vw-38px)] rounded-full bg-lightGreen p-7 opacity-95"
  onClick={[Function]}
>
  <img
    alt="plus"
    className="opacity-40 w-6"
    src="/assets/plus.png"
  />
</button>
`;
